From f2a611ce6cb0f86d9331641a804de6a507900db7 Mon Sep 17 00:00:00 2001
From: Tomasz Olszak <olszak.tomasz@gmail.com>
Date: Thu, 18 Jul 2013 20:45:47 +0000
Subject: [PATCH] Fixed mkspecs/devices/linux_device_post.conf for non-arm
 platforms.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Now arm specific -mfloat-abi flag is added to compiler flags
only for arm architecture in linux_arm_device_post.conf.

Change-Id: Ie77ac6e0717d9d1fd9c14e1d6a26e86f08ab418c
Reviewed-by: Jarosław Staniek <staniek@kde.org>
Reviewed-by: Oswald Buddenhagen <oswald.buddenhagen@digia.com>
Signed-off-by: Fatih Aşıcı <fatih.asici@gmail.com>
---
 mkspecs/devices/common/linux_arm_device_post.conf | 7 +++++++
 mkspecs/devices/common/linux_device_post.conf     | 6 ------
 mkspecs/devices/linux-imx53qsb-g++/qmake.conf     | 2 +-
 mkspecs/devices/linux-imx6-g++/qmake.conf         | 2 +-
 mkspecs/devices/linux-rasp-pi-g++/qmake.conf      | 2 +-
 mkspecs/devices/linux-tegra2-g++/qmake.conf       | 2 +-
 6 files changed, 11 insertions(+), 10 deletions(-)
 create mode 100644 mkspecs/devices/common/linux_arm_device_post.conf

diff --git a/mkspecs/devices/common/linux_arm_device_post.conf b/mkspecs/devices/common/linux_arm_device_post.conf
new file mode 100644
index 0000000..7ce4759
--- /dev/null
+++ b/mkspecs/devices/common/linux_arm_device_post.conf
@@ -0,0 +1,7 @@
+contains(DISTRO_OPTS, hard-float) {
+    COMPILER_FLAGS += -mfloat-abi=hard
+} else {
+    COMPILER_FLAGS += -mfloat-abi=softfp
+}
+
+include(linux_device_post.conf)
diff --git a/mkspecs/devices/common/linux_device_post.conf b/mkspecs/devices/common/linux_device_post.conf
index f8dbf76..548e75a 100644
--- a/mkspecs/devices/common/linux_device_post.conf
+++ b/mkspecs/devices/common/linux_device_post.conf
@@ -3,12 +3,6 @@ contains(DISTRO_OPTS, deb-multi-arch) {
                       -Wl,-rpath-link,$$[QT_SYSROOT]/lib/$${GCC_MACHINE_DUMP}
 }
 
-contains(DISTRO_OPTS, hard-float) {
-    COMPILER_FLAGS += -mfloat-abi=hard
-} else {
-    COMPILER_FLAGS += -mfloat-abi=softfp
-}
-
 QMAKE_CFLAGS       += $$COMPILER_FLAGS
 QMAKE_CXXFLAGS     += $$COMPILER_FLAGS
 
diff --git a/mkspecs/devices/linux-imx53qsb-g++/qmake.conf b/mkspecs/devices/linux-imx53qsb-g++/qmake.conf
index 0a5ed89..3a9766c 100644
--- a/mkspecs/devices/linux-imx53qsb-g++/qmake.conf
+++ b/mkspecs/devices/linux-imx53qsb-g++/qmake.conf
@@ -32,6 +32,6 @@ QMAKE_CXXFLAGS_RELEASE += $$IMX5_CFLAGS_RELEASE
 QMAKE_CFLAGS_DEBUG     += $$IMX5_CFLAGS
 QMAKE_CXXFLAGS_DEBUG   += $$IMX5_CFLAGS
 
-include(../common/linux_device_post.conf)
+include(../common/linux_arm_device_post.conf)
 
 load(qt_config)
diff --git a/mkspecs/devices/linux-imx6-g++/qmake.conf b/mkspecs/devices/linux-imx6-g++/qmake.conf
index 20f6d11..2b8dbf6 100644
--- a/mkspecs/devices/linux-imx6-g++/qmake.conf
+++ b/mkspecs/devices/linux-imx6-g++/qmake.conf
@@ -34,6 +34,6 @@ QMAKE_CXXFLAGS_RELEASE += $$IMX6_CFLAGS_RELEASE
 QMAKE_CFLAGS_DEBUG     += $$IMX6_CFLAGS
 QMAKE_CXXFLAGS_DEBUG   += $$IMX6_CFLAGS
 
-include(../common/linux_device_post.conf)
+include(../common/linux_arm_device_post.conf)
 
 load(qt_config)
diff --git a/mkspecs/devices/linux-rasp-pi-g++/qmake.conf b/mkspecs/devices/linux-rasp-pi-g++/qmake.conf
index d6fea47..5f923ad 100644
--- a/mkspecs/devices/linux-rasp-pi-g++/qmake.conf
+++ b/mkspecs/devices/linux-rasp-pi-g++/qmake.conf
@@ -42,6 +42,6 @@ QMAKE_CXXFLAGS          = $$QMAKE_CFLAGS
 EGLFS_PLATFORM_HOOKS_SOURCES = $$PWD/qeglfshooks_pi.cpp
 EGLFS_PLATFORM_HOOKS_LIBS = -lbcm_host
 
-include(../common/linux_device_post.conf)
+include(../common/linux_arm_device_post.conf)
 
 load(qt_config)
diff --git a/mkspecs/devices/linux-tegra2-g++/qmake.conf b/mkspecs/devices/linux-tegra2-g++/qmake.conf
index 1c7a8cc..320e1b8 100644
--- a/mkspecs/devices/linux-tegra2-g++/qmake.conf
+++ b/mkspecs/devices/linux-tegra2-g++/qmake.conf
@@ -26,6 +26,6 @@ TEGRA2_CFLAGS           = -mtune=cortex-a9 -march=armv7-a -mhard-float -mfloat-a
 QMAKE_CFLAGS           += $$TEGRA2_CFLAGS
 QMAKE_CXXFLAGS         += $$TEGRA2_CFLAGS
 
-include(../common/linux_device_post.conf)
+include(../common/linux_arm_device_post.conf)
 
 load(qt_config)
-- 
1.8.4.rc3

